comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -68,33 +62,26 @@ type resourceTags struct {
 	containerInstanceTags map[string]string
 }
 
-// NewCollector returns a new ecs collector provider and an error
-func NewCollector(deps dependencies) (workloadmeta.CollectorProvider, error) {
-	return workloadmeta.CollectorProvider{
-		Collector: &collector{
-			id:                           collectorID,
-			resourceTags:                 make(map[string]resourceTags),
-			seen:                         make(map[workloadmeta.EntityID]struct{}),
-			catalog:                      workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
-			config:                       deps.Config,
-			taskCollectionEnabled:        util.IsTaskCollectionEnabled(deps.Config),
-			taskCache:                    cache.New(deps.Config.GetDuration("ecs_task_cache_ttl"), 30*time.Second),
-			taskRateRPS:                  deps.Config.GetInt("ecs_task_collection_rate"),
-			taskRateBurst:                deps.Config.GetInt("ecs_task_collection_burst"),
-			metadataRetryInitialInterval: deps.Config.GetDuration("ecs_metadata_retry_initial_interval"),
-			metadataRetryMaxElapsedTime:  deps.Config.GetDuration("ecs_metadata_retry_max_elapsed_time"),
-			metadataRetryTimeoutFactor:   deps.Config.GetInt("ecs_metadata_retry_timeout_factor"),
-		},
+// NewCollector returns a new ecs collector
+func NewCollector(cfg config.Component) (wmcatalog.Collector, error) {
+	return &collector{
+		id:                           collectorID,
+		config:                       cfg,
+		resourceTags:                 make(map[string]resourceTags),
+		seen:                         make(map[workloadmeta.EntityID]struct{}),
+		catalog:                      workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
+		taskCollectionEnabled:        util.IsTaskCollectionEnabled(cfg),
+		taskCache:                    cache.New(cfg.GetDuration("ecs_task_cache_ttl"), 30*time.Second),
+		taskRateRPS:                  cfg.GetInt("ecs_task_collection_rate"),
+		taskRateBurst:                cfg.GetInt("ecs_task_collection_burst"),
+		metadataRetryInitialInterval: cfg.GetDuration("ecs_metadata_retry_initial_interval"),
+		metadataRetryMaxElapsedTime:  cfg.GetDuration("ecs_metadata_retry_max_elapsed_time"),
+		metadataRetryTimeoutFactor:   cfg.GetInt("ecs_metadata_retry_timeout_factor"),
 	}, nil
 }
 
-// GetFxOptions returns the FX framework options for the collector
-func GetFxOptions() fx.Option {
-	return fx.Provide(NewCollector)
-}
-
 func (c *collector) Start(ctx context.Context, store workloadmeta.Component) error {
-	if !pkgConfig.IsFeaturePresent(pkgConfig.ECSEC2) {
+	if !pkgconfig.IsFeaturePresent(pkgconfig.ECSEC2) {
```

Comment:
Nitpick: We could remove the usage of `pkgconfig` and use `pkg/config/env` to get the same outcome:

`!env.IsFeaturePresent(env.ECSEC2)`

Reply:
Ah, I didn't realize that they were aliased from the env package.